Release/2024 02 23

ludtkemorgan opened the pull request:
# Pull Request Template

## Issue Overview

This PR addresses [#389](https://github.com/bloom-housing/bloom/issues/3893)

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR pulls over the prisma changes from core and also has the majority of the changes specific to Doorway.

*Note* working through the following still so the PR isn't 100% ready, but can be looked at by the team to find any issues.

What is still remaining:
- ~Filtering on the listings page~
- ~Tests working in github actions~
- ~Updating the docker images for backend~
- Pull in the mandated account changes

## How Can This Be Tested/Reviewed?

This can only be tested locally and all flows on both public/partner sites should be tested (excluding the items in "what is still remaining" above)

Notes to test this locally:
- Make sure you have a `.env` file in api directory with the values from `.env.template`. You can use the same values you might have setup in core. But note there are a few values that are specific to Doorway such as the values to publish to s3. If you don't already have those values reach out to Morgan
- The best way to get setup is to do the following:
  - run `yarn install` from api directory
  - run `yarn prisma generate` from api directory
  - run `yarn db:reseed:with-external:dev` from api directory
  - run `yarn dev:all` from root directory (make sure you have also installed at root)

Specific areas that should be tested extra closely:
- Adding images to a listing (make sure that the aws flow works)
- Filtering on the public listings page. This was rewritten so making sure that no functionality on the listing page is broken


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

emilyjablonski commented:
`yarn setup:dev` doesn't work for me here, are you seeing something different?
Running seed command `ts-node prisma/seed.ts --environment development --jurisdictionName Bay Area` ...
TypeError [ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL]: Unexpected argument 'Area'. This command does not take positional arguments

emilyjablonski commented:
@ludtkemorgan That worked thank you! It could just be me, but as I'm poking around I'm noticing that the partners site is moving very slowly - it could be something unrelated, but just curious if others notice the same / wanted to flag

ColinBuyck commented:
<img width="1234" alt="Screenshot 2024-03-08 at 10 33 41 AM" src="https://github.com/metrotranscom/doorway/assets/53269332/b3885ab8-cc65-4c63-a765-2fea4ee90782">
Can't seem to get the requested changes email to send

ColinBuyck commented:
For documentation sake, I found that the listing image upload was erroring if the name was greater than or equal to 128 characters

ludtkemorgan commented:
> <img alt="Screenshot 2024-03-08 at 10 33 41 AM" width="1234" src="https://private-user-images.githubusercontent.com/53269332/311332325-b3885ab8-cc65-4c63-a765-2fea4ee90782.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5MjQ0MDIsIm5iZiI6MTcwOTkyNDEwMiwicGF0aCI6Ii81MzI2OTMzMi8zMTEzMzIzMjUtYjM4ODVhYjgtY2M2NS00YzYzLWE3NjUtMmZlYTRlZTkwNzgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA4VDE4NTUwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjZjQwNjI0ZWJlOGQzZTVjMTJjMTgxMWI5YzViMGNmOGVjOTYzYTc4NWQ0MDQ2MTNlMzYwYmE4ZDk2Y2Y3MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Sx40gSvxEIZOfhl9wJgWGWHXLTnKc0jLhh9ya67TzPU"> Can't seem to get the requested changes email to send

It was determined that this is also a bug in core. A PR has been made there to address it https://github.com/bloom-housing/bloom/pull/3934

YazeedLoonat commented on api/src/services/listing.service.ts:
```diff
@@ -229,6 +230,122 @@ export class ListingService implements OnModuleInit {
+  }> {
+    const onlyLettersPattern = /^[A-Za-z ]+$/;
+    const whereClauseArray = [];
+    if (params?.filter?.length) {
```
looks good!

YazeedLoonat reviewed (changes requested):
soem minor questions looks good overall!

ColinBuyck commented on api/src/services/user.service.ts:
```diff
@@ -0,0 +1,852 @@
+
+    // only update userRoles if something has changed
+    if (dto.userRoles && storedUser.userRoles) {
+      if (
```
Think we should include the recent security changes in this release

ludtkemorgan commented on api/Procfile:
```diff
@@ -0,0 +1 @@
+web: yarn start:prod
```
This isn't needed, but I didn't bother deleting any of the files for the core setup such as Heroku and circleCI files. That is similar to what we did with the old backend

ludtkemorgan commented:
@YazeedLoonat Your comments have been addressed. I also added a few more things since your last review:
- Pulled in the latest commits to main.
- Added the listing opportunity flow (govDelivery)
- Realized there was one more field needed on the combined listings table (reserved community types)

YazeedLoonat reviewed (approved):
LGTM
